Fix: Ensure focus visibility for “Show more” button in responsive view

ZallaxDev opened the pull request:
Adjusted the focus styling of the “Show more” button in the responsive and zoomed views to ensure that the element remains visibly highlighted when focused. This improves keyboard navigation and compliance with accessibility guidelines (BITV 9.2.4.7).
- Removed border assignment in the engaget state of the button in `_ui-component_metabar.scss`

Ticket [43958](https://mantis.ilias.de/view.php?id=43958)

ZallaxDev committed:
Fix: Ensure focus visibility for “Show more” button in responsive view

Adjusted the focus styling of the “Show more” button in the responsive and zoomed views to ensure that the element remains visibly highlighted when focused.
This improves keyboard navigation and compliance with accessibility guidelines (BITV 9.2.4.7).
Reference: Ticket #43958.

catenglaender commented:
Thank you @ZallaxDev and Surlabs for looking into this. :slightly_smiling_face: Accessibility fixes like this one are very important for many institutions - especially in such a common, visible spot. It's very much appreciated! :pray: 

If you can spend a little more time on this issue, I would be really happy about a solution that goes beyond a patch and is more solid and future proof:
1. All engaged states should have their own variables, so designers can just replace them at the top of the document to get the look they want for their skin e.g.
```SCSS
		&.engaged {
			box-shadow: $il-mainbar-entry-engaged-shadow;
			background-color: $il-metabar-entry-engaged-bg-color;
			color: $il-metabar-entry-engaged-bg-color;
			outline-color: transparent; // not sure why this is here... try without it
		}
```
This might mean passing some less specific variables to the super specific variables e.g. `$il-metabar-entry-engaged-shadow: $il-metabar-shadow-right`. It's a bit of overhead, but it makes each component a bit more modular and encapsulated.

2. And now for the special sauce: If necessary, call the mixin for the focus on the engaged state again. It does produce a bit more CSS code but it explicitly marks the places where (re-)setting the focus is required.
```SCSS
		&.engaged {
                    // [...] all other code
                    @include f.il-focus-border-only();
		}
```
In doing so, you specifically override the border properties for the focus visible state. This way skin designers can use their own border color for the engaged state and the focus state is set specific enough to override it for sure.

Let me know if you have any questions here or on Discord. :slightly_smiling_face:  Thanks again for your participation tackling these accessibility details!

All the best,
Ferdinand

ZallaxDev committed:
Fix: restore border and improve focus handling

- Restored the previously removed border to maintain consistent visual boundaries.
- Added `@include f.il-focus-border-only()` to apply focus styling in a cleaner, reusable way aligned with accessibility guidelines.
- Removed the unnecessary `outline-color` property, testing confirmed it doesn’t affect the expected result.

ZallaxDev commented:
Thank you very much, Ferdinand, for taking the time to provide such a detailed and thoughtful explanation.

I really appreciate the guidance on making this solution more robust and future-proof. I’ve implemented the changes following your recommendations. Reapplying the focus mixin within the .engaged state helped us better understand how the code works and how to approach better implementations in the future.
I also removed the outline-color property. After our tests, it doesn’t seem to negatively affect the expected result, so we believe it’s fine to omit it.

Your detailed feedback is extremely valuable and helps us continue improving the quality and maintainability of our solutions. Thanks again for your time and for keeping accessibility a priority across the platform🙏

Best regards,
Daniel.

catenglaender commented:
Your kind words and diligent work is very much appreciated. I know these nitpicks and standards can sometimes be a little bit annoying, but over all we will get a more robust system and less follow-up issues in the long run. Thank you very much for your understanding and contribution! :slightly_smiling_face:
